apps/src/templates/testPages/WebLabTest.jsx

```diff
@@ -0,0 +1,179 @@
+            any questions.
+          </p>
+        </div>
+      );
+    }
+
+    return (
+      <p>
+        <strong>You passed all checks! No further action is needed.</strong>
+      </p>
+    );
+  };
+
+  render() {
+    return (
+      <div id="main-content" className="container">
+        <div className="row">
+          <div className="col-md-6">
+            <div className="form-group">
+              <h1>Web Lab Network Support Check</h1>
+              <p>
+                Use this page to test your school's network support for our Web
+                Lab modules. Web Lab communicates with domains which may be
+                blocked by some school systems’ networks or device policies. If
+                the network check is unsuccessful, you will need to reach out to
+                your school's IT department to update your school's firewall
+                settings or device policies.
+              </p>
+              <p>
+                When checking network support, try to meet as many of these
+                criteria as possible:
+              </p>
+              <li>Use your school's internet connection</li>
+              <li>Use a student computer</li>
+              <li>Log into the computer with a student account</li>
+              <br />
+              <p>
+                Note: You do not need to be logged in to Code.org for this test
+                to be successful.
+              </p>
+              <label htmlFor="connect">
+                Click "Connect" to check whether your school's network supports
+                the Web Lab tool
+              </label>
+              <div id="status-table">
+                <table>
+                  <tbody>
+                    <tr>
+                      <th>Check</th>
+                      <th>Status</th>
+                    </tr>
+                    <tr>
+                      <td>
+                        Connected to
+                        <code>https://downloads.computinginthecore.org</code>
+                      </td>
+                      <td className="computinginthecore">
+                        {this.renderTestStatus(STATUS_COMPUTING_IN_THE_CORE)}
+                      </td>
+                    </tr>
+                    <tr>
+                      <td>
+                        Connected to <code>https://codeprojects.org</code>
+                      </td>
+                      <td className="codeprojects">
+                        {this.renderTestStatus(STATUS_CODE_PROJECTS)}
+                      </td>
+                    </tr>
+                  </tbody>
+                </table>
+              </div>
+              <div>
+                <button
+                  type="submit"
+                  id="connect"
+                  className="btn btn-primary"
+                  onClick={this.runWebLabTest}
+                >
+                  Connect
+                </button>
```

Comment:
Given that the button modifies the table above, would it make sense to move the button above the table?

Reply:
I'm unsure what convention dictates here and am happy to defer to it. I thought it was a plus that the user has to scroll past the table (possibly registering its contents) before hitting `Connect`, so maybe they know what it said before it gets modified in-line, but I am happy to move it up too.

Maybe moving the button is a change to make in the next PR (when the entire page moves to studio.code.org) so that while the CSA and Web Lab test pages are living side-by-side at `educate/websockets` and `educate/weblab`, they share styling?

Reply:
Sure thing.